# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libGLESv2/entry_points_gles_1_0_autogen.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libGLESv2/entry_points_gles_1_0_autogen.cpp
--- a/gfx/angle/checkout/src/libGLESv2/entry_points_gles_1_0_autogen.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libGLESv2/entry_points_gles_1_0_autogen.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -12,21 +12,22 @@
 
 #include "libANGLE/Context.h"
 #include "libANGLE/Context.inl.h"
+#include "libANGLE/capture_gles_1_0_autogen.h"
+#include "libANGLE/entry_points_utils.h"
 #include "libANGLE/validationES1.h"
-#include "libGLESv2/entry_points_utils.h"
 #include "libGLESv2/global_state.h"
 
 namespace gl
 {
 void GL_APIENTRY AlphaFunc(GLenum func, GLfloat ref)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum func = 0x%X, GLfloat ref = %f)", func, ref);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         AlphaTestFunc funcPacked = FromGLenum<AlphaTestFunc>(func);
+        ANGLE_CAPTURE(AlphaFunc, context, funcPacked, ref);
         if (context->skipValidation() || ValidateAlphaFunc(context, funcPacked, ref))
         {
             context->alphaFunc(funcPacked, ref);
@@ -36,13 +37,13 @@
 
 void GL_APIENTRY AlphaFuncx(GLenum func, GLfixed ref)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum func = 0x%X, GLfixed ref = 0x%X)", func, ref);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         AlphaTestFunc funcPacked = FromGLenum<AlphaTestFunc>(func);
+        ANGLE_CAPTURE(AlphaFuncx, context, funcPacked, ref);
         if (context->skipValidation() || ValidateAlphaFuncx(context, funcPacked, ref))
         {
             context->alphaFuncx(funcPacked, ref);
@@ -52,13 +53,13 @@
 
 void GL_APIENTRY ClearColorx(GLfixed red, GLfixed green, GLfixed blue, GLfixed alpha)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed red = 0x%X, GLfixed green = 0x%X, GLfixed blue = 0x%X, GLfixed alpha = 0x%X)",
           red, green, blue, alpha);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ClearColorx, context, red, green, blue, alpha);
         if (context->skipValidation() || ValidateClearColorx(context, red, green, blue, alpha))
         {
             context->clearColorx(red, green, blue, alpha);
@@ -68,12 +69,12 @@
 
 void GL_APIENTRY ClearDepthx(GLfixed depth)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed depth = 0x%X)", depth);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ClearDepthx, context, depth);
         if (context->skipValidation() || ValidateClearDepthx(context, depth))
         {
             context->clearDepthx(depth);
@@ -83,12 +84,12 @@
 
 void GL_APIENTRY ClientActiveTexture(GLenum texture)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum texture = 0x%X)", texture);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ClientActiveTexture, context, texture);
         if (context->skipValidation() || ValidateClientActiveTexture(context, texture))
         {
             context->clientActiveTexture(texture);
@@ -98,12 +99,12 @@
 
 void GL_APIENTRY ClipPlanef(GLenum p, const GLfloat *eqn)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum p = 0x%X, const GLfloat *eqn = 0x%016" PRIxPTR ")", p, (uintptr_t)eqn);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ClipPlanef, context, p, eqn);
         if (context->skipValidation() || ValidateClipPlanef(context, p, eqn))
         {
             context->clipPlanef(p, eqn);
@@ -113,13 +114,13 @@
 
 void GL_APIENTRY ClipPlanex(GLenum plane, const GLfixed *equation)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum plane = 0x%X, const GLfixed *equation = 0x%016" PRIxPTR ")", plane,
           (uintptr_t)equation);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(ClipPlanex, context, plane, equation);
         if (context->skipValidation() || ValidateClipPlanex(context, plane, equation))
         {
             context->clipPlanex(plane, equation);
@@ -129,13 +130,13 @@
 
 void GL_APIENTRY Color4f(GLfloat red, GLfloat green, GLfloat blue, GLfloat alpha)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfloat red = %f, GLfloat green = %f, GLfloat blue = %f, GLfloat alpha = %f)", red,
           green, blue, alpha);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Color4f, context, red, green, blue, alpha);
         if (context->skipValidation() || ValidateColor4f(context, red, green, blue, alpha))
         {
             context->color4f(red, green, blue, alpha);
@@ -145,13 +146,13 @@
 
 void GL_APIENTRY Color4ub(GLubyte red, GLubyte green, GLubyte blue, GLubyte alpha)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLubyte red = %d, GLubyte green = %d, GLubyte blue = %d, GLubyte alpha = %d)", red,
           green, blue, alpha);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Color4ub, context, red, green, blue, alpha);
         if (context->skipValidation() || ValidateColor4ub(context, red, green, blue, alpha))
         {
             context->color4ub(red, green, blue, alpha);
@@ -161,13 +162,13 @@
 
 void GL_APIENTRY Color4x(GLfixed red, GLfixed green, GLfixed blue, GLfixed alpha)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed red = 0x%X, GLfixed green = 0x%X, GLfixed blue = 0x%X, GLfixed alpha = 0x%X)",
           red, green, blue, alpha);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Color4x, context, red, green, blue, alpha);
         if (context->skipValidation() || ValidateColor4x(context, red, green, blue, alpha))
         {
             context->color4x(red, green, blue, alpha);
@@ -177,7 +178,6 @@
 
 void GL_APIENTRY ColorPointer(GLint size, GLenum type, GLsizei stride, const void *pointer)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLint size = %d, GLenum type = 0x%X, GLsizei stride = %d, const void *pointer = "
         "0x%016" PRIxPTR ")",
@@ -187,6 +187,7 @@
     if (context)
     {
         VertexAttribType typePacked = FromGLenum<VertexAttribType>(type);
+        ANGLE_CAPTURE(ColorPointer, context, size, typePacked, stride, pointer);
         if (context->skipValidation() ||
             ValidateColorPointer(context, size, typePacked, stride, pointer))
         {
@@ -197,12 +198,12 @@
 
 void GL_APIENTRY DepthRangex(GLfixed n, GLfixed f)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed n = 0x%X, GLfixed f = 0x%X)", n, f);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(DepthRangex, context, n, f);
         if (context->skipValidation() || ValidateDepthRangex(context, n, f))
         {
             context->depthRangex(n, f);
@@ -212,13 +213,13 @@
 
 void GL_APIENTRY DisableClientState(GLenum array)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum array = 0x%X)", array);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         ClientVertexArrayType arrayPacked = FromGLenum<ClientVertexArrayType>(array);
+        ANGLE_CAPTURE(DisableClientState, context, arrayPacked);
         if (context->skipValidation() || ValidateDisableClientState(context, arrayPacked))
         {
             context->disableClientState(arrayPacked);
@@ -228,13 +229,13 @@
 
 void GL_APIENTRY EnableClientState(GLenum array)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum array = 0x%X)", array);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         ClientVertexArrayType arrayPacked = FromGLenum<ClientVertexArrayType>(array);
+        ANGLE_CAPTURE(EnableClientState, context, arrayPacked);
         if (context->skipValidation() || ValidateEnableClientState(context, arrayPacked))
         {
             context->enableClientState(arrayPacked);
@@ -244,12 +245,12 @@
 
 void GL_APIENTRY Fogf(GLenum pname, GLfloat param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLfloat param = %f)", pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Fogf, context, pname, param);
         if (context->skipValidation() || ValidateFogf(context, pname, param))
         {
             context->fogf(pname, param);
@@ -259,13 +260,13 @@
 
 void GL_APIENTRY Fogfv(GLenum pname, const GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, const GLfloat *params = 0x%016" PRIxPTR ")", pname,
           (uintptr_t)params);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Fogfv, context, pname, params);
         if (context->skipValidation() || ValidateFogfv(context, pname, params))
         {
             context->fogfv(pname, params);
@@ -275,12 +276,12 @@
 
 void GL_APIENTRY Fogx(GLenum pname, GLfixed param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLfixed param = 0x%X)", pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Fogx, context, pname, param);
         if (context->skipValidation() || ValidateFogx(context, pname, param))
         {
             context->fogx(pname, param);
@@ -290,13 +291,13 @@
 
 void GL_APIENTRY Fogxv(GLenum pname, const GLfixed *param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, const GLfixed *param = 0x%016" PRIxPTR ")", pname,
           (uintptr_t)param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Fogxv, context, pname, param);
         if (context->skipValidation() || ValidateFogxv(context, pname, param))
         {
             context->fogxv(pname, param);
@@ -306,7 +307,6 @@
 
 void GL_APIENTRY Frustumf(GLfloat l, GLfloat r, GLfloat b, GLfloat t, GLfloat n, GLfloat f)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLfloat l = %f, GLfloat r = %f, GLfloat b = %f, GLfloat t = %f, GLfloat n = %f, GLfloat "
         "f = %f)",
@@ -315,6 +315,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Frustumf, context, l, r, b, t, n, f);
         if (context->skipValidation() || ValidateFrustumf(context, l, r, b, t, n, f))
         {
             context->frustumf(l, r, b, t, n, f);
@@ -324,7 +325,6 @@
 
 void GL_APIENTRY Frustumx(GLfixed l, GLfixed r, GLfixed b, GLfixed t, GLfixed n, GLfixed f)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLfixed l = 0x%X, GLfixed r = 0x%X, GLfixed b = 0x%X, GLfixed t = 0x%X, GLfixed n = "
         "0x%X, GLfixed f = 0x%X)",
@@ -333,6 +333,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Frustumx, context, l, r, b, t, n, f);
         if (context->skipValidation() || ValidateFrustumx(context, l, r, b, t, n, f))
         {
             context->frustumx(l, r, b, t, n, f);
@@ -342,13 +343,13 @@
 
 void GL_APIENTRY GetClipPlanef(GLenum plane, GLfloat *equation)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum plane = 0x%X, GLfloat *equation = 0x%016" PRIxPTR ")", plane,
           (uintptr_t)equation);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetClipPlanef, context, plane, equation);
         if (context->skipValidation() || ValidateGetClipPlanef(context, plane, equation))
         {
             context->getClipPlanef(plane, equation);
@@ -358,13 +359,13 @@
 
 void GL_APIENTRY GetClipPlanex(GLenum plane, GLfixed *equation)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum plane = 0x%X, GLfixed *equation = 0x%016" PRIxPTR ")", plane,
           (uintptr_t)equation);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetClipPlanex, context, plane, equation);
         if (context->skipValidation() || ValidateGetClipPlanex(context, plane, equation))
         {
             context->getClipPlanex(plane, equation);
@@ -374,12 +375,12 @@
 
 void GL_APIENTRY GetFixedv(GLenum pname, GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLfixed *params = 0x%016" PRIxPTR ")", pname, (uintptr_t)params);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetFixedv, context, pname, params);
         if (context->skipValidation() || ValidateGetFixedv(context, pname, params))
         {
             context->getFixedv(pname, params);
@@ -389,7 +390,6 @@
 
 void GL_APIENTRY GetLightfv(GLenum light, GLenum pname, GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum light = 0x%X, GLenum pname = 0x%X, GLfloat *params = 0x%016" PRIxPTR ")", light,
           pname, (uintptr_t)params);
 
@@ -397,6 +397,7 @@
     if (context)
     {
         LightParameter pnamePacked = FromGLenum<LightParameter>(pname);
+        ANGLE_CAPTURE(GetLightfv, context, light, pnamePacked, params);
         if (context->skipValidation() || ValidateGetLightfv(context, light, pnamePacked, params))
         {
             context->getLightfv(light, pnamePacked, params);
@@ -406,7 +407,6 @@
 
 void GL_APIENTRY GetLightxv(GLenum light, GLenum pname, GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum light = 0x%X, GLenum pname = 0x%X, GLfixed *params = 0x%016" PRIxPTR ")", light,
           pname, (uintptr_t)params);
 
@@ -414,6 +414,7 @@
     if (context)
     {
         LightParameter pnamePacked = FromGLenum<LightParameter>(pname);
+        ANGLE_CAPTURE(GetLightxv, context, light, pnamePacked, params);
         if (context->skipValidation() || ValidateGetLightxv(context, light, pnamePacked, params))
         {
             context->getLightxv(light, pnamePacked, params);
@@ -423,7 +424,6 @@
 
 void GL_APIENTRY GetMaterialfv(GLenum face, GLenum pname, GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum face = 0x%X, GLenum pname = 0x%X, GLfloat *params = 0x%016" PRIxPTR ")", face,
           pname, (uintptr_t)params);
 
@@ -431,6 +431,7 @@
     if (context)
     {
         MaterialParameter pnamePacked = FromGLenum<MaterialParameter>(pname);
+        ANGLE_CAPTURE(GetMaterialfv, context, face, pnamePacked, params);
         if (context->skipValidation() || ValidateGetMaterialfv(context, face, pnamePacked, params))
         {
             context->getMaterialfv(face, pnamePacked, params);
@@ -440,7 +441,6 @@
 
 void GL_APIENTRY GetMaterialxv(GLenum face, GLenum pname, GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum face = 0x%X, GLenum pname = 0x%X, GLfixed *params = 0x%016" PRIxPTR ")", face,
           pname, (uintptr_t)params);
 
@@ -448,6 +448,7 @@
     if (context)
     {
         MaterialParameter pnamePacked = FromGLenum<MaterialParameter>(pname);
+        ANGLE_CAPTURE(GetMaterialxv, context, face, pnamePacked, params);
         if (context->skipValidation() || ValidateGetMaterialxv(context, face, pnamePacked, params))
         {
             context->getMaterialxv(face, pnamePacked, params);
@@ -457,12 +458,12 @@
 
 void GL_APIENTRY GetPointerv(GLenum pname, void **params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, void **params = 0x%016" PRIxPTR ")", pname, (uintptr_t)params);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(GetPointerv, context, pname, params);
         if (context->skipValidation() || ValidateGetPointerv(context, pname, params))
         {
             context->getPointerv(pname, params);
@@ -472,7 +473,6 @@
 
 void GL_APIENTRY GetTexEnvfv(GLenum target, GLenum pname, GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLfloat *params = 0x%016" PRIxPTR ")",
           target, pname, (uintptr_t)params);
 
@@ -481,6 +481,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(GetTexEnvfv, context, targetPacked, pnamePacked, params);
         if (context->skipValidation() ||
             ValidateGetTexEnvfv(context, targetPacked, pnamePacked, params))
         {
@@ -491,7 +492,6 @@
 
 void GL_APIENTRY GetTexEnviv(GLenum target, GLenum pname, GLint *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLint *params = 0x%016" PRIxPTR ")", target,
           pname, (uintptr_t)params);
 
@@ -500,6 +500,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(GetTexEnviv, context, targetPacked, pnamePacked, params);
         if (context->skipValidation() ||
             ValidateGetTexEnviv(context, targetPacked, pnamePacked, params))
         {
@@ -510,7 +511,6 @@
 
 void GL_APIENTRY GetTexEnvxv(GLenum target, GLenum pname, GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLfixed *params = 0x%016" PRIxPTR ")",
           target, pname, (uintptr_t)params);
 
@@ -519,6 +519,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(GetTexEnvxv, context, targetPacked, pnamePacked, params);
         if (context->skipValidation() ||
             ValidateGetTexEnvxv(context, targetPacked, pnamePacked, params))
         {
@@ -529,7 +530,6 @@
 
 void GL_APIENTRY GetTexParameterxv(GLenum target, GLenum pname, GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLfixed *params = 0x%016" PRIxPTR ")",
           target, pname, (uintptr_t)params);
 
@@ -537,6 +537,7 @@
     if (context)
     {
         TextureType targetPacked = FromGLenum<TextureType>(target);
+        ANGLE_CAPTURE(GetTexParameterxv, context, targetPacked, pname, params);
         if (context->skipValidation() ||
             ValidateGetTexParameterxv(context, targetPacked, pname, params))
         {
@@ -547,12 +548,12 @@
 
 void GL_APIENTRY LightModelf(GLenum pname, GLfloat param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLfloat param = %f)", pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LightModelf, context, pname, param);
         if (context->skipValidation() || ValidateLightModelf(context, pname, param))
         {
             context->lightModelf(pname, param);
@@ -562,13 +563,13 @@
 
 void GL_APIENTRY LightModelfv(GLenum pname, const GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, const GLfloat *params = 0x%016" PRIxPTR ")", pname,
           (uintptr_t)params);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LightModelfv, context, pname, params);
         if (context->skipValidation() || ValidateLightModelfv(context, pname, params))
         {
             context->lightModelfv(pname, params);
@@ -578,12 +579,12 @@
 
 void GL_APIENTRY LightModelx(GLenum pname, GLfixed param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLfixed param = 0x%X)", pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LightModelx, context, pname, param);
         if (context->skipValidation() || ValidateLightModelx(context, pname, param))
         {
             context->lightModelx(pname, param);
@@ -593,13 +594,13 @@
 
 void GL_APIENTRY LightModelxv(GLenum pname, const GLfixed *param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, const GLfixed *param = 0x%016" PRIxPTR ")", pname,
           (uintptr_t)param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LightModelxv, context, pname, param);
         if (context->skipValidation() || ValidateLightModelxv(context, pname, param))
         {
             context->lightModelxv(pname, param);
@@ -609,13 +610,13 @@
 
 void GL_APIENTRY Lightf(GLenum light, GLenum pname, GLfloat param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum light = 0x%X, GLenum pname = 0x%X, GLfloat param = %f)", light, pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         LightParameter pnamePacked = FromGLenum<LightParameter>(pname);
+        ANGLE_CAPTURE(Lightf, context, light, pnamePacked, param);
         if (context->skipValidation() || ValidateLightf(context, light, pnamePacked, param))
         {
             context->lightf(light, pnamePacked, param);
@@ -625,7 +626,6 @@
 
 void GL_APIENTRY Lightfv(GLenum light, GLenum pname, const GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum light = 0x%X, GLenum pname = 0x%X, const GLfloat *params = 0x%016" PRIxPTR ")",
           light, pname, (uintptr_t)params);
 
@@ -633,6 +633,7 @@
     if (context)
     {
         LightParameter pnamePacked = FromGLenum<LightParameter>(pname);
+        ANGLE_CAPTURE(Lightfv, context, light, pnamePacked, params);
         if (context->skipValidation() || ValidateLightfv(context, light, pnamePacked, params))
         {
             context->lightfv(light, pnamePacked, params);
@@ -642,13 +643,13 @@
 
 void GL_APIENTRY Lightx(GLenum light, GLenum pname, GLfixed param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum light = 0x%X, GLenum pname = 0x%X, GLfixed param = 0x%X)", light, pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         LightParameter pnamePacked = FromGLenum<LightParameter>(pname);
+        ANGLE_CAPTURE(Lightx, context, light, pnamePacked, param);
         if (context->skipValidation() || ValidateLightx(context, light, pnamePacked, param))
         {
             context->lightx(light, pnamePacked, param);
@@ -658,7 +659,6 @@
 
 void GL_APIENTRY Lightxv(GLenum light, GLenum pname, const GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum light = 0x%X, GLenum pname = 0x%X, const GLfixed *params = 0x%016" PRIxPTR ")",
           light, pname, (uintptr_t)params);
 
@@ -666,6 +666,7 @@
     if (context)
     {
         LightParameter pnamePacked = FromGLenum<LightParameter>(pname);
+        ANGLE_CAPTURE(Lightxv, context, light, pnamePacked, params);
         if (context->skipValidation() || ValidateLightxv(context, light, pnamePacked, params))
         {
             context->lightxv(light, pnamePacked, params);
@@ -675,12 +676,12 @@
 
 void GL_APIENTRY LineWidthx(GLfixed width)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed width = 0x%X)", width);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LineWidthx, context, width);
         if (context->skipValidation() || ValidateLineWidthx(context, width))
         {
             context->lineWidthx(width);
@@ -690,12 +691,12 @@
 
 void GL_APIENTRY LoadIdentity()
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("()");
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LoadIdentity, context);
         if (context->skipValidation() || ValidateLoadIdentity(context))
         {
             context->loadIdentity();
@@ -705,12 +706,12 @@
 
 void GL_APIENTRY LoadMatrixf(const GLfloat *m)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(const GLfloat *m = 0x%016" PRIxPTR ")", (uintptr_t)m);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LoadMatrixf, context, m);
         if (context->skipValidation() || ValidateLoadMatrixf(context, m))
         {
             context->loadMatrixf(m);
@@ -720,12 +721,12 @@
 
 void GL_APIENTRY LoadMatrixx(const GLfixed *m)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(const GLfixed *m = 0x%016" PRIxPTR ")", (uintptr_t)m);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(LoadMatrixx, context, m);
         if (context->skipValidation() || ValidateLoadMatrixx(context, m))
         {
             context->loadMatrixx(m);
@@ -735,13 +736,13 @@
 
 void GL_APIENTRY LogicOp(GLenum opcode)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum opcode = 0x%X)", opcode);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         LogicalOperation opcodePacked = FromGLenum<LogicalOperation>(opcode);
+        ANGLE_CAPTURE(LogicOp, context, opcodePacked);
         if (context->skipValidation() || ValidateLogicOp(context, opcodePacked))
         {
             context->logicOp(opcodePacked);
@@ -751,13 +752,13 @@
 
 void GL_APIENTRY Materialf(GLenum face, GLenum pname, GLfloat param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum face = 0x%X, GLenum pname = 0x%X, GLfloat param = %f)", face, pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         MaterialParameter pnamePacked = FromGLenum<MaterialParameter>(pname);
+        ANGLE_CAPTURE(Materialf, context, face, pnamePacked, param);
         if (context->skipValidation() || ValidateMaterialf(context, face, pnamePacked, param))
         {
             context->materialf(face, pnamePacked, param);
@@ -767,7 +768,6 @@
 
 void GL_APIENTRY Materialfv(GLenum face, GLenum pname, const GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum face = 0x%X, GLenum pname = 0x%X, const GLfloat *params = 0x%016" PRIxPTR ")",
           face, pname, (uintptr_t)params);
 
@@ -775,6 +775,7 @@
     if (context)
     {
         MaterialParameter pnamePacked = FromGLenum<MaterialParameter>(pname);
+        ANGLE_CAPTURE(Materialfv, context, face, pnamePacked, params);
         if (context->skipValidation() || ValidateMaterialfv(context, face, pnamePacked, params))
         {
             context->materialfv(face, pnamePacked, params);
@@ -784,13 +785,13 @@
 
 void GL_APIENTRY Materialx(GLenum face, GLenum pname, GLfixed param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum face = 0x%X, GLenum pname = 0x%X, GLfixed param = 0x%X)", face, pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         MaterialParameter pnamePacked = FromGLenum<MaterialParameter>(pname);
+        ANGLE_CAPTURE(Materialx, context, face, pnamePacked, param);
         if (context->skipValidation() || ValidateMaterialx(context, face, pnamePacked, param))
         {
             context->materialx(face, pnamePacked, param);
@@ -800,7 +801,6 @@
 
 void GL_APIENTRY Materialxv(GLenum face, GLenum pname, const GLfixed *param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum face = 0x%X, GLenum pname = 0x%X, const GLfixed *param = 0x%016" PRIxPTR ")",
           face, pname, (uintptr_t)param);
 
@@ -808,6 +808,7 @@
     if (context)
     {
         MaterialParameter pnamePacked = FromGLenum<MaterialParameter>(pname);
+        ANGLE_CAPTURE(Materialxv, context, face, pnamePacked, param);
         if (context->skipValidation() || ValidateMaterialxv(context, face, pnamePacked, param))
         {
             context->materialxv(face, pnamePacked, param);
@@ -817,13 +818,13 @@
 
 void GL_APIENTRY MatrixMode(GLenum mode)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum mode = 0x%X)", mode);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         MatrixType modePacked = FromGLenum<MatrixType>(mode);
+        ANGLE_CAPTURE(MatrixMode, context, modePacked);
         if (context->skipValidation() || ValidateMatrixMode(context, modePacked))
         {
             context->matrixMode(modePacked);
@@ -833,12 +834,12 @@
 
 void GL_APIENTRY MultMatrixf(const GLfloat *m)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(const GLfloat *m = 0x%016" PRIxPTR ")", (uintptr_t)m);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(MultMatrixf, context, m);
         if (context->skipValidation() || ValidateMultMatrixf(context, m))
         {
             context->multMatrixf(m);
@@ -848,12 +849,12 @@
 
 void GL_APIENTRY MultMatrixx(const GLfixed *m)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(const GLfixed *m = 0x%016" PRIxPTR ")", (uintptr_t)m);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(MultMatrixx, context, m);
         if (context->skipValidation() || ValidateMultMatrixx(context, m))
         {
             context->multMatrixx(m);
@@ -863,13 +864,13 @@
 
 void GL_APIENTRY MultiTexCoord4f(GLenum target, GLfloat s, GLfloat t, GLfloat r, GLfloat q)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLfloat s = %f, GLfloat t = %f, GLfloat r = %f, GLfloat q = %f)",
           target, s, t, r, q);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(MultiTexCoord4f, context, target, s, t, r, q);
         if (context->skipValidation() || ValidateMultiTexCoord4f(context, target, s, t, r, q))
         {
             context->multiTexCoord4f(target, s, t, r, q);
@@ -879,7 +880,6 @@
 
 void GL_APIENTRY MultiTexCoord4x(GLenum texture, GLfixed s, GLfixed t, GLfixed r, GLfixed q)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLenum texture = 0x%X, GLfixed s = 0x%X, GLfixed t = 0x%X, GLfixed r = 0x%X, GLfixed q = "
         "0x%X)",
@@ -888,6 +888,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(MultiTexCoord4x, context, texture, s, t, r, q);
         if (context->skipValidation() || ValidateMultiTexCoord4x(context, texture, s, t, r, q))
         {
             context->multiTexCoord4x(texture, s, t, r, q);
@@ -897,12 +898,12 @@
 
 void GL_APIENTRY Normal3f(GLfloat nx, GLfloat ny, GLfloat nz)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfloat nx = %f, GLfloat ny = %f, GLfloat nz = %f)", nx, ny, nz);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Normal3f, context, nx, ny, nz);
         if (context->skipValidation() || ValidateNormal3f(context, nx, ny, nz))
         {
             context->normal3f(nx, ny, nz);
@@ -912,12 +913,12 @@
 
 void GL_APIENTRY Normal3x(GLfixed nx, GLfixed ny, GLfixed nz)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed nx = 0x%X, GLfixed ny = 0x%X, GLfixed nz = 0x%X)", nx, ny, nz);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Normal3x, context, nx, ny, nz);
         if (context->skipValidation() || ValidateNormal3x(context, nx, ny, nz))
         {
             context->normal3x(nx, ny, nz);
@@ -927,7 +928,6 @@
 
 void GL_APIENTRY NormalPointer(GLenum type, GLsizei stride, const void *pointer)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum type = 0x%X, GLsizei stride = %d, const void *pointer = 0x%016" PRIxPTR ")",
           type, stride, (uintptr_t)pointer);
 
@@ -935,6 +935,7 @@
     if (context)
     {
         VertexAttribType typePacked = FromGLenum<VertexAttribType>(type);
+        ANGLE_CAPTURE(NormalPointer, context, typePacked, stride, pointer);
         if (context->skipValidation() ||
             ValidateNormalPointer(context, typePacked, stride, pointer))
         {
@@ -945,7 +946,6 @@
 
 void GL_APIENTRY Orthof(GLfloat l, GLfloat r, GLfloat b, GLfloat t, GLfloat n, GLfloat f)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLfloat l = %f, GLfloat r = %f, GLfloat b = %f, GLfloat t = %f, GLfloat n = %f, GLfloat "
         "f = %f)",
@@ -954,6 +954,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Orthof, context, l, r, b, t, n, f);
         if (context->skipValidation() || ValidateOrthof(context, l, r, b, t, n, f))
         {
             context->orthof(l, r, b, t, n, f);
@@ -963,7 +964,6 @@
 
 void GL_APIENTRY Orthox(GLfixed l, GLfixed r, GLfixed b, GLfixed t, GLfixed n, GLfixed f)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLfixed l = 0x%X, GLfixed r = 0x%X, GLfixed b = 0x%X, GLfixed t = 0x%X, GLfixed n = "
         "0x%X, GLfixed f = 0x%X)",
@@ -972,6 +972,7 @@
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Orthox, context, l, r, b, t, n, f);
         if (context->skipValidation() || ValidateOrthox(context, l, r, b, t, n, f))
         {
             context->orthox(l, r, b, t, n, f);
@@ -981,13 +982,13 @@
 
 void GL_APIENTRY PointParameterf(GLenum pname, GLfloat param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLfloat param = %f)", pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         PointParameter pnamePacked = FromGLenum<PointParameter>(pname);
+        ANGLE_CAPTURE(PointParameterf, context, pnamePacked, param);
         if (context->skipValidation() || ValidatePointParameterf(context, pnamePacked, param))
         {
             context->pointParameterf(pnamePacked, param);
@@ -997,7 +998,6 @@
 
 void GL_APIENTRY PointParameterfv(GLenum pname, const GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, const GLfloat *params = 0x%016" PRIxPTR ")", pname,
           (uintptr_t)params);
 
@@ -1005,6 +1005,7 @@
     if (context)
     {
         PointParameter pnamePacked = FromGLenum<PointParameter>(pname);
+        ANGLE_CAPTURE(PointParameterfv, context, pnamePacked, params);
         if (context->skipValidation() || ValidatePointParameterfv(context, pnamePacked, params))
         {
             context->pointParameterfv(pnamePacked, params);
@@ -1014,13 +1015,13 @@
 
 void GL_APIENTRY PointParameterx(GLenum pname, GLfixed param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, GLfixed param = 0x%X)", pname, param);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         PointParameter pnamePacked = FromGLenum<PointParameter>(pname);
+        ANGLE_CAPTURE(PointParameterx, context, pnamePacked, param);
         if (context->skipValidation() || ValidatePointParameterx(context, pnamePacked, param))
         {
             context->pointParameterx(pnamePacked, param);
@@ -1030,7 +1031,6 @@
 
 void GL_APIENTRY PointParameterxv(GLenum pname, const GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum pname = 0x%X, const GLfixed *params = 0x%016" PRIxPTR ")", pname,
           (uintptr_t)params);
 
@@ -1038,6 +1038,7 @@
     if (context)
     {
         PointParameter pnamePacked = FromGLenum<PointParameter>(pname);
+        ANGLE_CAPTURE(PointParameterxv, context, pnamePacked, params);
         if (context->skipValidation() || ValidatePointParameterxv(context, pnamePacked, params))
         {
             context->pointParameterxv(pnamePacked, params);
@@ -1047,12 +1048,12 @@
 
 void GL_APIENTRY PointSize(GLfloat size)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfloat size = %f)", size);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(PointSize, context, size);
         if (context->skipValidation() || ValidatePointSize(context, size))
         {
             context->pointSize(size);
@@ -1062,12 +1063,12 @@
 
 void GL_APIENTRY PointSizex(GLfixed size)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed size = 0x%X)", size);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(PointSizex, context, size);
         if (context->skipValidation() || ValidatePointSizex(context, size))
         {
             context->pointSizex(size);
@@ -1077,12 +1078,12 @@
 
 void GL_APIENTRY PolygonOffsetx(GLfixed factor, GLfixed units)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed factor = 0x%X, GLfixed units = 0x%X)", factor, units);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(PolygonOffsetx, context, factor, units);
         if (context->skipValidation() || ValidatePolygonOffsetx(context, factor, units))
         {
             context->polygonOffsetx(factor, units);
@@ -1092,12 +1093,12 @@
 
 void GL_APIENTRY PopMatrix()
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("()");
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(PopMatrix, context);
         if (context->skipValidation() || ValidatePopMatrix(context))
         {
             context->popMatrix();
@@ -1107,12 +1108,12 @@
 
 void GL_APIENTRY PushMatrix()
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("()");
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(PushMatrix, context);
         if (context->skipValidation() || ValidatePushMatrix(context))
         {
             context->pushMatrix();
@@ -1122,12 +1123,12 @@
 
 void GL_APIENTRY Rotatef(GLfloat angle, GLfloat x, GLfloat y, GLfloat z)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfloat angle = %f, GLfloat x = %f, GLfloat y = %f, GLfloat z = %f)", angle, x, y, z);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Rotatef, context, angle, x, y, z);
         if (context->skipValidation() || ValidateRotatef(context, angle, x, y, z))
         {
             context->rotatef(angle, x, y, z);
@@ -1137,13 +1138,13 @@
 
 void GL_APIENTRY Rotatex(GLfixed angle, GLfixed x, GLfixed y, GLfixed z)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed angle = 0x%X, GLfixed x = 0x%X, GLfixed y = 0x%X, GLfixed z = 0x%X)", angle, x,
           y, z);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Rotatex, context, angle, x, y, z);
         if (context->skipValidation() || ValidateRotatex(context, angle, x, y, z))
         {
             context->rotatex(angle, x, y, z);
@@ -1153,12 +1154,12 @@
 
 void GL_APIENTRY SampleCoveragex(GLclampx value, GLboolean invert)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLclampx value = 0x%X, GLboolean invert = %u)", value, invert);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(SampleCoveragex, context, value, invert);
         if (context->skipValidation() || ValidateSampleCoveragex(context, value, invert))
         {
             context->sampleCoveragex(value, invert);
@@ -1168,12 +1169,12 @@
 
 void GL_APIENTRY Scalef(GLfloat x, GLfloat y, GLfloat z)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfloat x = %f, GLfloat y = %f, GLfloat z = %f)", x, y, z);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Scalef, context, x, y, z);
         if (context->skipValidation() || ValidateScalef(context, x, y, z))
         {
             context->scalef(x, y, z);
@@ -1183,12 +1184,12 @@
 
 void GL_APIENTRY Scalex(GLfixed x, GLfixed y, GLfixed z)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed x = 0x%X, GLfixed y = 0x%X, GLfixed z = 0x%X)", x, y, z);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Scalex, context, x, y, z);
         if (context->skipValidation() || ValidateScalex(context, x, y, z))
         {
             context->scalex(x, y, z);
@@ -1198,13 +1199,13 @@
 
 void GL_APIENTRY ShadeModel(GLenum mode)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum mode = 0x%X)", mode);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
         ShadingModel modePacked = FromGLenum<ShadingModel>(mode);
+        ANGLE_CAPTURE(ShadeModel, context, modePacked);
         if (context->skipValidation() || ValidateShadeModel(context, modePacked))
         {
             context->shadeModel(modePacked);
@@ -1214,7 +1215,6 @@
 
 void GL_APIENTRY TexCoordPointer(GLint size, GLenum type, GLsizei stride, const void *pointer)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLint size = %d, GLenum type = 0x%X, GLsizei stride = %d, const void *pointer = "
         "0x%016" PRIxPTR ")",
@@ -1224,6 +1224,7 @@
     if (context)
     {
         VertexAttribType typePacked = FromGLenum<VertexAttribType>(type);
+        ANGLE_CAPTURE(TexCoordPointer, context, size, typePacked, stride, pointer);
         if (context->skipValidation() ||
             ValidateTexCoordPointer(context, size, typePacked, stride, pointer))
         {
@@ -1234,7 +1235,6 @@
 
 void GL_APIENTRY TexEnvf(GLenum target, GLenum pname, GLfloat param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLfloat param = %f)", target, pname, param);
 
     Context *context = GetValidGlobalContext();
@@ -1242,6 +1242,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(TexEnvf, context, targetPacked, pnamePacked, param);
         if (context->skipValidation() || ValidateTexEnvf(context, targetPacked, pnamePacked, param))
         {
             context->texEnvf(targetPacked, pnamePacked, param);
@@ -1251,7 +1252,6 @@
 
 void GL_APIENTRY TexEnvfv(GLenum target, GLenum pname, const GLfloat *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, const GLfloat *params = 0x%016" PRIxPTR ")",
           target, pname, (uintptr_t)params);
 
@@ -1260,6 +1260,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(TexEnvfv, context, targetPacked, pnamePacked, params);
         if (context->skipValidation() ||
             ValidateTexEnvfv(context, targetPacked, pnamePacked, params))
         {
@@ -1270,7 +1271,6 @@
 
 void GL_APIENTRY TexEnvi(GLenum target, GLenum pname, GLint param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLint param = %d)", target, pname, param);
 
     Context *context = GetValidGlobalContext();
@@ -1278,6 +1278,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(TexEnvi, context, targetPacked, pnamePacked, param);
         if (context->skipValidation() || ValidateTexEnvi(context, targetPacked, pnamePacked, param))
         {
             context->texEnvi(targetPacked, pnamePacked, param);
@@ -1287,7 +1288,6 @@
 
 void GL_APIENTRY TexEnviv(GLenum target, GLenum pname, const GLint *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, const GLint *params = 0x%016" PRIxPTR ")",
           target, pname, (uintptr_t)params);
 
@@ -1296,6 +1296,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(TexEnviv, context, targetPacked, pnamePacked, params);
         if (context->skipValidation() ||
             ValidateTexEnviv(context, targetPacked, pnamePacked, params))
         {
@@ -1306,7 +1307,6 @@
 
 void GL_APIENTRY TexEnvx(GLenum target, GLenum pname, GLfixed param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLfixed param = 0x%X)", target, pname,
           param);
 
@@ -1315,6 +1315,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(TexEnvx, context, targetPacked, pnamePacked, param);
         if (context->skipValidation() || ValidateTexEnvx(context, targetPacked, pnamePacked, param))
         {
             context->texEnvx(targetPacked, pnamePacked, param);
@@ -1324,7 +1325,6 @@
 
 void GL_APIENTRY TexEnvxv(GLenum target, GLenum pname, const GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, const GLfixed *params = 0x%016" PRIxPTR ")",
           target, pname, (uintptr_t)params);
 
@@ -1333,6 +1333,7 @@
     {
         TextureEnvTarget targetPacked   = FromGLenum<TextureEnvTarget>(target);
         TextureEnvParameter pnamePacked = FromGLenum<TextureEnvParameter>(pname);
+        ANGLE_CAPTURE(TexEnvxv, context, targetPacked, pnamePacked, params);
         if (context->skipValidation() ||
             ValidateTexEnvxv(context, targetPacked, pnamePacked, params))
         {
@@ -1343,7 +1344,6 @@
 
 void GL_APIENTRY TexParameterx(GLenum target, GLenum pname, GLfixed param)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, GLfixed param = 0x%X)", target, pname,
           param);
 
@@ -1351,6 +1351,7 @@
     if (context)
     {
         TextureType targetPacked = FromGLenum<TextureType>(target);
+        ANGLE_CAPTURE(TexParameterx, context, targetPacked, pname, param);
         if (context->skipValidation() || ValidateTexParameterx(context, targetPacked, pname, param))
         {
             context->texParameterx(targetPacked, pname, param);
@@ -1360,7 +1361,6 @@
 
 void GL_APIENTRY TexParameterxv(GLenum target, GLenum pname, const GLfixed *params)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLenum target = 0x%X, GLenum pname = 0x%X, const GLfixed *params = 0x%016" PRIxPTR ")",
           target, pname, (uintptr_t)params);
 
@@ -1368,6 +1368,7 @@
     if (context)
     {
         TextureType targetPacked = FromGLenum<TextureType>(target);
+        ANGLE_CAPTURE(TexParameterxv, context, targetPacked, pname, params);
         if (context->skipValidation() ||
             ValidateTexParameterxv(context, targetPacked, pname, params))
         {
@@ -1378,12 +1379,12 @@
 
 void GL_APIENTRY Translatef(GLfloat x, GLfloat y, GLfloat z)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfloat x = %f, GLfloat y = %f, GLfloat z = %f)", x, y, z);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Translatef, context, x, y, z);
         if (context->skipValidation() || ValidateTranslatef(context, x, y, z))
         {
             context->translatef(x, y, z);
@@ -1393,12 +1394,12 @@
 
 void GL_APIENTRY Translatex(GLfixed x, GLfixed y, GLfixed z)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT("(GLfixed x = 0x%X, GLfixed y = 0x%X, GLfixed z = 0x%X)", x, y, z);
 
     Context *context = GetValidGlobalContext();
     if (context)
     {
+        ANGLE_CAPTURE(Translatex, context, x, y, z);
         if (context->skipValidation() || ValidateTranslatex(context, x, y, z))
         {
             context->translatex(x, y, z);
@@ -1408,7 +1409,6 @@
 
 void GL_APIENTRY VertexPointer(GLint size, GLenum type, GLsizei stride, const void *pointer)
 {
-    ANGLE_SCOPED_GLOBAL_LOCK();
     EVENT(
         "(GLint size = %d, GLenum type = 0x%X, GLsizei stride = %d, const void *pointer = "
         "0x%016" PRIxPTR ")",
@@ -1418,6 +1418,7 @@
     if (context)
     {
         VertexAttribType typePacked = FromGLenum<VertexAttribType>(type);
+        ANGLE_CAPTURE(VertexPointer, context, size, typePacked, stride, pointer);
         if (context->skipValidation() ||
             ValidateVertexPointer(context, size, typePacked, stride, pointer))
         {